src/components/ArticleVote/ArticleVote.jsx

```diff
@@ -0,0 +1,55 @@
+import { patchArticle } from "../../utils/utils";
+import { useState } from "react";
+
+export default function ArticleVote({ setArticleContent, article }) {
+  const [vote, setVote] = useState("");
+
+  function handleUpVote() {
+    setArticleContent((prevContent) => {
+      return prevContent.map((article) => {
+        return { ...article, votes: article.votes + 1 };
+      });
+    });
+
+    patchArticle(article, 1)
+      .then(() => {
+        setVote("Upvoted");
+      })
+      .catch(() => {
+        setVote("err");
+      });
+  }
+
+  function handleDownVote() {
+    setArticleContent((prevContent) => {
+      return prevContent.map((article) => {
+        return { ...article, votes: article.votes - 1 };
+      });
+    });
+
+    patchArticle(article, -1)
+      .then(() => {
+        setVote("Downvoted");
+      })
+      .catch(() => {
+        setVote("err");
+      });
+  }
+
+  return (
+    <>
+      {vote ? (
+        vote === "err" ? (
+          <p id="msg">Error voting on this article</p>
```

Comment:
this is good. But the whole Article Card changing to red and button disappearing regardless of voting outcome is a bit confusing as to what is happening. Colour change is permanent and alway red regardless of outcome.  Might be worth rethinking how this looks.